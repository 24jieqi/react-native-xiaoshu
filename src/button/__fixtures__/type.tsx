/**
 * title: 按钮类型
 * description: 按钮有五种类型：主按钮（默认）、朦胧按钮、边框按钮、幽灵按钮和链接按钮。
 */

import React, { memo } from 'react'

import {
  Button,
  Card,
  Space,
  Divider,
} from '@fruits-chain/react-native-xiaoshu'

const onPress = () => {
  console.log('点击按钮 => ', new Date().getTime())
}

const ButtonType: React.FC = () => {
  return (
    <Card title="按钮类型" square>
      <Space>
        <Space>
          <Button
            type="primary"
            text="primary 主按钮&点击事件防抖动"
            onPress={onPress}
            onPressDebounceWait={500}
          />
          <Button
            type="hazy"
            text="hazy 朦胧按钮"
            onPress={onPress}
            onPressDebounceWait={300}
          />
          <Button type="outline" text="outline 边框按钮" onPress={onPress} />
          <Button type="ghost" text="ghost 幽灵按钮" onPress={onPress} />
          <Button type="link" text="link 链接按钮" onPress={onPress} />
        </Space>

        <Divider>·</Divider>

        <Space direction="horizontal" wrap>
          <Button type="primary" text="primary" onPress={onPress} />
          <Button type="hazy" text="hazy" onPress={onPress} />
          <Button type="outline" text="outline" onPress={onPress} />
          <Button type="ghost" text="ghost" onPress={onPress} />
          <Button type="link" text="link" onPress={onPress} />
        </Space>
      </Space>
    </Card>
  )
}

export default memo(ButtonType)
