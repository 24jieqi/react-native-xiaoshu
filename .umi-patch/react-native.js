export * from 'react-native-web'

export const InputAccessoryView = () => {}
