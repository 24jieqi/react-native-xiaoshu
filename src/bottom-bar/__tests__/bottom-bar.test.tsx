import React from 'react'
import type { ViewStyle } from 'react-native'
import { Text, StyleSheet, Platform } from 'react-native'
// import RCTDeviceEventEmitter from 'react-native/Libraries/EventEmitter/RCTDeviceEventEmitter'
import { SafeAreaProvider } from 'react-native-safe-area-context'
// eslint-disable-next-line import/no-named-as-default-member,import/namespace,import/no-named-as-default
import { create } from 'react-test-renderer'

import BottomBar from '..'
import { customRender, act } from '../../__tests__/test-utils'

describe('Blank', () => {
  it('render snapshot', () => {
    const tree = create(
      <SafeAreaProvider>
        <BottomBar>
          <Text>Content</Text>
        </BottomBar>
      </SafeAreaProvider>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('android:keyboardDidXxx', () => {
    // mockPlatform('android', 23)

    Platform.OS = 'android'

    customRender(
      <BottomBar height={60} testID="height">
        <Text>Content</Text>
      </BottomBar>,
    )

    act(() => {
      // RCTDeviceEventEmitter.DeviceEventEmitter.emit('keyboardDidShow', {})
    })

    act(() => {
      // DeviceEventEmitter.emit('keyboardDidHide', {})
    })
  })

  it('height', () => {
    const { getByTestId } = customRender(
      <BottomBar height={60} testID="height">
        <Text>Content</Text>
      </BottomBar>,
    )

    const style = StyleSheet.flatten<ViewStyle>(
      getByTestId('height').props.style,
    )

    expect(style.height).toEqual(60)
  })
})
