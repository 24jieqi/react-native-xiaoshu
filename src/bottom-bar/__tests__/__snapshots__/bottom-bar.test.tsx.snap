// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blank render snapshot 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderTopColor": "#EFF3F9",
      "borderTopWidth": 1,
      "height": 50,
      "overflow": "hidden",
      "paddingBottom": 0,
    }
  }
>
  <Text>
    Content
  </Text>
</View>
`;
